lib/Model.php

```diff
@@ -1025,4 +1052,16 @@ public static function key () {
 		$o = new $class;
 		return $o->key;
 	}
+	
+	/**
+	 * Get the type of primary key for this model.
+	 * If type is specified, a boolean comparison is returned.
+	 */
+	private function _key_type ($type = false) {
```

Comment:
Model defaults $key to 'id', but I'm seeing `Model does not have a primary key(s).` errors in the unit tests now. Not sure all the errors are related to that, but maybe you can shed some light on what I'm seeing.

Here's the phpunit output on Model:

```
There were 7 failures:

1) ModelTest::test_put
Failed asserting that two strings are equal.
--- Expected
+++ Actual
@@ @@
-'qwerty'
+'foobar'

/Users/lux/projects/elefant/www/tests/ModelTest.php:251

2) ModelTest::test_get
Failed asserting that two objects are equal.
--- Expected
+++ Actual
@@ @@
 Qwerty Object (
     'key' => 'foo'
     'table' => 'qwerty'
     'keyval' => false
     'data' => Array (
+        'foo' => 'asdf'
+        'bar' => 'foobar'
     )
     'fields' => Array ()
-    'error' => 'Model does not have a primary key(s).'
+    'error' => false
     'is_new' => false
     'query_fields' => '*'
     'query_order' => ''
     'query_group' => ''
     'query_filters' => Array ()
     'query_params' => Array ()
     'query_having' => Array ()
     'query_from' => false
     'verify' => Array ()
     'failed' => Array ()
 )

/Users/lux/projects/elefant/www/tests/ModelTest.php:257

3) ModelTest::test_fetch_assoc
Failed asserting that two arrays are equal.
--- Expected
+++ Actual
@@ @@
 Array (
-    'asdf' => 'qwerty'
+    'asdf' => 'foobar'
 )

/Users/lux/projects/elefant/www/tests/ModelTest.php:264

4) ModelTest::test_fetch_field
Failed asserting that two arrays are equal.
--- Expected
+++ Actual
@@ @@
 Array (
-    0 => 'qwerty'
+    0 => 'foobar'
 )

/Users/lux/projects/elefant/www/tests/ModelTest.php:270

5) ModelTest::test_remove
Failed asserting that two objects are equal.
--- Expected
+++ Actual
@@ @@
         'foo' => 'asdf'
-        'bar' => 'qwerty'
+        'bar' => 'foobar'
     )
     'fields' => Array ()
     'error' => false
     'is_new' => false
     'query_fields' => '*'
     'query_order' => ''
     'query_group' => ''
     'query_filters' => Array ()
     'query_params' => Array ()
     'query_having' => Array ()
     'query_from' => false
     'verify' => Array ()
     'failed' => Array ()
 )

/Users/lux/projects/elefant/www/tests/ModelTest.php:278

6) ModelTest::test_batch
Failed asserting that two strings are equal.
--- Expected
+++ Actual
@@ @@
-'Joe'
+'One'

/Users/lux/projects/elefant/www/tests/ModelTest.php:333

7) ModelTest::test_references
Failed asserting that two strings are equal.
--- Expected
+++ Actual
@@ @@
-'One'
+'Joe'

/Users/lux/projects/elefant/www/tests/ModelTest.php:366
                                       
FAILURES!                              
Tests: 19, Assertions: 62, Failures: 7.
```